model/artifact/artifact_file.go

```diff
@@ -84,7 +97,39 @@ func GetAllArtifacts(tasks []TaskIDAndExecution) ([]File, error) {
 		}
 	}
 	files := []File{}
+	catcher := grip.NewBasicCatcher()
 	for _, artifact := range artifacts {
+		for i := range artifact.Files {
+			if artifact.Files[i].Visibility == "signed" {
+				if artifact.Files[i].AwsSecret == "" || artifact.Files[i].AwsKey == "" || artifact.Files[i].Bucket == "" || artifact.Files[i].FileKey == "" {
+					err = errors.New("error presigning the url for artifact")
+					catcher.Add(err)
```

Comment:
you can do catcher.New() to avoid the extra call. 

also it seems like we'd want to continue here.